### PR TITLE
Fix 4.2 and added features for thumbnail of wikipedia 

### PR DESCRIPTION
**Pull request by descovi**
Support for build with Travis (from pull request of  @hugoabonizio - https://github.com/dejan/auto_html/pull/150 ) and support of thumbnail from wikipedia (example http://upload.wikimedia.org/wikipedia/commons/thumb/9/98/SPietro_al_Monte_basilica.JPG/220px-SPietro_al_Monte_basilica.JPG)

### CODE REVIEW

File: lib/auto_html/filters/image.rb [hunk 1]
```diff
@@ -10,7 +10,9 @@ def paragraph(text)
   r = Redcarpet::Markdown.new(NoParagraphRenderer)
   alt = options[:alt]
   options[:proxy] ||= ""
-  text.gsub(/(?<!src=")https?:\/\/.+?\.(jpg|jpeg|bmp|gif|png)(\?\S+)?/i) do |match|
+  # original /(?<!src=")https?:\/\/.+?\.(jpg|jpeg|bmp|gif|png)(\?\S+)?/i
```
Comment (dgilperez) [inline lib/auto_html/filters/image.rb]: Can you please clean this?

File: lib/auto_html/filters/sanitize.rb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-    else
-      Rails::Html::WhiteListSanitizer
-    end
+require 'action_controller'
+require 'cgi'
 
-  sanitizer_class.new.sanitize(text, options)
-end
+AutoHtml.add_filter(:sanitize).with({}) do |text, options|
+  if Rails::VERSION::MAJOR == 4 and Rails::VERSION::MINOR >= 2
```
Comment (dgilperez) [inline lib/auto_html/filters/sanitize.rb]: I actually prefer to check if `defined?(HTML::WhiteListSanitizer)` than to a fixed version, I think it's more consistent. Can you please remove this change from the PR or split it into two?

File: test/unit/filters/image_test.rb [hunk 3]
```diff
@@ -52,6 +52,11 @@ def test_transform8
     assert_equal 'Which do you prefer, this one <img src="http://www.lockhartfineart.com/images/Rio_Grande_Frost.JPG" alt=""/>, or this one <img src="http://rors.org/images/rails.png" alt=""/>?', result
   end
 
+  def test_transform9
```
Comment (dgilperez) [inline test/unit/filters/image_test.rb:55]: Great!

File: test/unit/filters/sanitize_test.rb [hunk 4]
```diff
@@ -24,7 +24,12 @@ def test_trasform4
 
   def test_trasform5
     result = auto_html("<a rel='nofollow'>test</div>") { sanitize :tags => %w(a), :attributes => %w(href)}
-    assert_equal "<a>test", result 
+     if Rails::VERSION::MAJOR == 4 and Rails::VERSION::MINOR >= 2
+      # Rails >=4.2
```
Comment (dgilperez) [inline test/unit/filters/sanitize_test.rb]: You can remove this comment

File: lib/auto_html/filters/image.rb (hunk 1 above)
Comment (descovi) [inline lib/auto_html/filters/image.rb]: :+1:

**Comment by dgilperez** [comment]
Hey @simonini, I'm so sorry about the delay here, I just forgot about this PR 😞  

Thanks for the work, merging it now. 

I'm tidying the gem up a bit more to prepare it for Rails 5 (which I don't know yet if it works)

**Pull request merged**